app/models/concerns/hostext/fact_data.rb

```diff
@@ -0,0 +1,58 @@
+module Hostext
+  module FactData
+    extend ActiveSupport::Concern
+
+    def set_reported_data(parser)
+      set_data(:boot_time, :boot_timestamp, parser, ->(value) { Time.at value })
+      set_data(:virtual, :virtual, parser)
+      set_data(:ram, :ram, parser, ->(value) { format_host_memory value })
+      set_data(:sockets, :sockets, parser)
+      set_data(:cores, :cores, parser)
+    end
+
+    def cores
+      get_data :cores
+    end
+
+    def virtual
+      get_data :virtual
+    end
+
+    def sockets
+      get_data :sockets
+    end
+
+    def ram
+      get_data :ram
+    end
+
+    def uptime_seconds
+      get_data(:boot_time, ->(value) { value.nil? ? nil : Time.zone.now.to_i - value.to_i })
+    end
+
+    def reported_data_facet
+      self.reported_data || self.build_reported_data
+    end
+
+    private
+
+    def get_data(attr_name, format_lambda = ->(value) { value })
```

Comment:
`format_lambda` should default to nil and we should return value if not specified, otherwise call it, or even better specific the transformation as a block and call it with `yield` (small performance gain).

Reply:
Maybe we should name the method `get_reported_data`?

Reply:
yeah, I agree, it's included into a huge object, better be safe, I guess the same for `set_data` below